### ISSUE TITLE
MADR File Requires Two Paragraphs in Context And Problem Statement

### ISSUE THREAD

**Issue by Jololol** [joshrotenberg/mdbook-lint#441]
**Version**
mdbook-lint 0.14.4 (CLI, lint subcommand)

**Summary**
When using MADR documents, I get linting errors pointing to the YAML front matter, which disappear when making unobvious changes to the MADR document.
This may be related to #406 

**Reproduction**
Use the config and sample MADR to test.  The warnings depicted below disappear if I do the following changes to the MADR:
* Duplicate the Context and Problem Statement so there are two paragraphs
* Delete the entire Considered Options section

**Config**
```toml
enabled-rules=["MD004"]
[MD004]
style = "asterisk" # Use * for list, part of MADR standard
```
**Sample MADR File**
```markdown
---
status: accepted
date: 2026-07-20
decision-makers:
  - DM 1
  - DM 2
consulted:
  - First Consult
informed:
  - First Informed
---
# ADR-0002: Adopt a CONTRIBUTING.md

## Context and Problem Statement

Contribution rules (commit message format, MR requirements, review expectations) are currently
tribal knowledge. New contributors learn them by having MRs rejected. How do we make these
rules discoverable and enforceable?

## Considered Options

* Adopt a `CONTRIBUTING.md` at the repository root
* Document rules in confluence
* Leave rules implicit and enforce via review
* Extra item
```
**Output**
```
warning[MD004]: Inconsistent list style: expected '*' but found '-'
  --> docs/adr/0002-contributing-guidelines.md:11:1
     |
  11 | ---
     | ^^^ ul-style
```

**Comment by joshrotenberg** [comment]
Likely the same root cause as #406 ("Standard rules (MD041, MD022, MD007) false-positive on YAML frontmatter in 0.14.4"), which explicitly noted the misparsed front-matter block context "cascades into further false positives (MD032, MD004)". This looks like that predicted MD004 cascade, either not covered by the #406 fix or reintroduced since. The paragraph-count / Considered-Options sensitivity is consistent with a line-offset-dependent parser state bug rather than anything content-specific.

**Comment by joshrotenberg** [comment]
ready: Confirmed as the MD004 cascade #406 predicted: #430 added Document::frontmatter_ast_offset and applied it to MD041/MD022/MD007, but MD004 still indexes document.lines with the unshifted AST line, so it reads the marker from inside the frontmatter (line 11 is the closing `---`). Add the offset as MD022 does; the autofix was rewriting that delimiter to `***`. Draft PR #445.
